assets/js/util/tracking/createTrackEvent.js

```diff
@@ -64,6 +64,10 @@ export default function createTrackEvent(
 			dimension4: global.GOOGLESITEKIT_VERSION || '',
 			dimension5: Array.from( enabledFeatures ).join( ',' ),
 			dimension6: activeModules.join( ',' ),
+			// Custom dimension to differentiate between authenticated and non authenticated users.
+			dimension7: global._googlesitekitUserData?.user?.authenticated
```

Comment:
I realize this was done once before but we should avoid reaching for globals directly here. [We use a number of globals already](https://github.com/google/site-kit-wp/blob/460a21e240c91f51f6a22ca17b1ba67b8162236e/assets/js/util/tracking/index.js#L24-L31) but these are passed through in the `config`, so let's take the same approach here.

Would you please do the same with the `currentUserRoles` above for consistency?

We still reference `console` and `_gaUserPrefs` on the global below but even then we use the injected `_global`, but these are the only exceptions we should need.

Also – please add/update the tests for `trackEvent` to ensure this dimension is covered.

Reply:
I've also updated the use of `GOOGLESITEKIT_VERSION` as well.